ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -284,7 +284,7 @@ public final int doPhase2Abort ()
 
 		org.omg.CosTransactions.Status s = get_status();
 
-		if ((s == org.omg.CosTransactions.Status.StatusCommitted)
+		if ((s == org.omg.CosTransactions.Status.StatusCommitted && preparedList.size() == 0)
```

Comment:
Well, the new state transaction is to run from COMMITTED through the ABORTING code rather than just return ABORTED.

As you will see, COMMITTED -> ABORTED was actually already possible (at least to an external caller): https://github.com/jbosstm/narayana/blob/master/ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java#L298